core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -769,6 +785,25 @@ private[spark] class TaskSetManager(
     }
   }
 
+  def setTaskRecordsAndRunTime(
+      info: TaskInfo,
+      result: DirectTaskResult[_]): Unit = {
+    var records = 0L
+    var runTime = 0L
+    result.accumUpdates.foreach { a =>
+      if (a.name == Some(shuffleRead.RECORDS_READ) ||
+        a.name == Some(input.RECORDS_READ)) {
+        val acc = a.asInstanceOf[LongAccumulator]
+        records += acc.value
+      } else if (a.name == Some(InternalAccumulator.EXECUTOR_RUN_TIME)) {
+        val acc = a.asInstanceOf[LongAccumulator]
+        runTime = acc.value
+      }
+    }
+    info.setRecords(records)
+    info.setRunTime(runTime)
```

Comment:
Same here.. I think we centralize the calculation of this stuff into `InefficientTaskCalculator`.